src/App/src/Middleware/ContentNegociationMiddleware.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Api\App\Middleware;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Laminas\Http\Response;
+use Mezzio\Router\RouteResult;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Dot\AnnotatedServices\Annotation\Inject;
+
+class ContentNegociationMiddleware implements MiddlewareInterface
+{
+
+    /**
+     * @Inject({"config.content-negotiation"})
+     */
+    public function __construct(private array $config)
+    {
+    }
+
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ): ResponseInterface {
+        $response = $handler->handle($request);
```

Comment:
I would use this line as it was in your first draft you showcased:

1. Call it only inside this if:

> if (!$routeResult instanceof RouteResult || $routeResult->isFailure()) {

2. The again, before this line:

> $responseContentType = $response->getHeaderLine('Content-Type') ?? "";

Because we should not execute the rest of the pipeline if either `checkAccept` or `checkContentType` fail.